src/brush.js

```diff
@@ -319,11 +313,23 @@ function brush(dim) {
       return this;
     },
     emit: function(type) {
-      customEvent(new BrushEvent(brush, type, dim.output(this.state.selection)), listeners.apply, listeners, [type, this.that, this.args]);
+      var dispatch = listeners.copy(),
```

Comment:
Is this a defensive copy? It looks like previously we did not make a copy.

Reply:
Ah I don't know. I copied this from https://github.com/d3/d3-drag/blob/f91e6bc297f04e652cd2262b7e56e948ac36c556/src/drag.js#L120 ; maybe it's not needed in this case though.

Reply:
That’s done exclusively at the start of the gesture so that you can register listeners for a specific gesture (rather than on every emitted event).

https://github.com/d3/d3-drag/blob/master/README.md#event_on

> Equivalent to drag.on, but only applies to the current drag gesture. Before the drag gesture starts, a copy of the current drag event listeners is made. This copy is bound to the current drag gesture and modified by event.on. This is useful for temporary listeners that only receive events for the current drag gesture.

We could do that for d3-brush, too, if you think it’s useful.

Reply:
Yes. 1) it looks cleaner 2) it will be better for maintenance to have the same approach in the three modules (drag, zoom, brush). I'll try to do this on the multitouch branch.

Reply:
added in https://github.com/d3/d3-brush/pull/71/commits/2cf062cf4b8da9f6246c7762ef54f9c2b117af66